ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -284,10 +296,10 @@ private void handleQuery(final SessionAndRequest info, final Query query) {
     queryPublisher.start(ksqlEngine, exec, configured, streamSubscriber);
```

Comment:
We have to pass the `serviceContext` to the `StreamPublisher#subscribe()` method somehow. 

Look at the method, it calls `ksqlEngine.execute(query)` which uses an internal `ServiceContext`. The correct impersonation execution would be `ksqlEngine.execute(serviceContext, query)`.

Reply:
Done. Thanks for the catch.